packages/@ember/-internals/runtime/lib/system/core_object.js

```diff
@@ -211,17 +195,9 @@ function initialize(obj, properties) {
   @public
 */
 class CoreObject {
-  static _initFactory(factory) {
-    factoryMap.set(this, factory);
-  }
-
-  constructor(passedFromCreate) {
-    // pluck off factory
-    let initFactory = factoryMap.get(this.constructor);
-    if (initFactory !== undefined) {
-      factoryMap.delete(this.constructor);
-      FACTORY_FOR.set(this, initFactory);
-    }
+  constructor(owner) {
+    // pluck off factory and set it as the instance factory
+    this[OWNER] = owner;
```

Comment:
This seems odd to me, why can't this use `setOwner`?

Reply:
Also the comment seems wrong, we are plucking the owner not the factory/instance factory.

Reply:
`setOwner` has to set both `OWNER` and `LEGACY_OWNER` for backwards compatibility (the `ember-m3` use case, for instance). `LEGACY_OWNER` is enumerable, so setting it would add an enumerable property to the object. I'll add a comment to clarify